### DISCUSSION TITLE
Opinions and Ideas for adding ElevenLabs extra tts options

### DISCUSSION THREAD

**Discussion by aymanmostafa11** [SubtitleEdit/subtitleedit#8928] [Ideas]
Hello Everyone, @niksedk amazing app 
  
I have added the extra ElevenLabs options in a [personal fork](https://github.com/aymanmostafa11/subtitleedit) previously and was hoping to get it merged into the official repo, I just need some opinions on
## Placement: 
I've done it like 
<img width=400 src=https://github.com/user-attachments/assets/531816b2-6be8-471d-a634-42ad73136584/><br>
but this hides the `Language` option when it gets displayed
<img width=400 src=https://github.com/user-attachments/assets/3266924c-8400-4346-8748-fd0de039cc25/> <br>
I don't want to do major changes in the application's sizes and placements but also couldn't get both to fit together

## Character Count
 I thought having an idea of how much a file will consume from the character quota was good insight to know before starting the generation so I've implemented it [here](https://github.com/SubtitleEdit/subtitleedit/commit/75cc14615adb20a95a7c02b84e59defe619ac925) like this to prevent recalculation each time the character count is requested

Any ideas, opinions and suggestions are very welcome so please don't hesitate to share your thoughts am more than happy to discuss them

**Comment by niksedk** [comment]
OK, I've added the `Stability` and `Similarity` with values between 0-100:

![image](https://github.com/user-attachments/assets/478d9ad5-1c2f-4264-a5e9-ace9e849b89b)

After reading a bit, I think we do not need more than these two settings.
https://elevenlabs.io/docs/speech-synthesis/voice-settings

Perhaps you could make make an info button or link label, next to the `Generate speech from text` button, that displays character count (and maybe more)?

**Reply by aymanmostafa11** [reply]
Thanks for the reply and for adding the new options 
Regarding the other options, although the documentation says they have subtle effects but newer models could possibly have more emphasis on them and since we're already adding the options,  I think it would be useful having them there for the sake of completeness

For the [context](https://elevenlabs.io/docs/api-reference/how-to-use-request-stitching) we found it useful in some cases, what's your thoughts on it?

Finally I believe I can do the info part, any other info you think should be included?

**Comment by natika1** [comment]
Did you considered AI language recognition ? Or choosing a voice will be always manual? Just thinking :)
